lnwallet/channel.go

```diff
@@ -1328,12 +1345,44 @@ func (lc *LightningChannel) fetchCommitmentView(remoteChain bool,
 		numHTLCs++
 	}
 
+	// Initiate feePerKw to the last committed fee for this chain.
+	feePerKw := commitChain.tail().feePerKw
+
+	// Check if any fee updates have taken place since that last commitment.
+	if lc.channelState.IsInitiator {
```

Comment:
I think this can be simplified substantially. 

Instead of setting the commitment fee here within `fetchCommitmentView`, I'd instead set it above when either receiving, or creating a new commitment. That extra context will, I think, allow us to shed the extra variable that's currently being used. 

Here's a rough sketch of the logic:
  * If sending a commitment:
       * If initiator and pending fee update
           * use that pending fee
       * else
            * use the commitment fee
  * If receiving a commitment:
       * If not initiator and pending fee update
           * use that pending fee
       * else
           * use the commitment fee

I _think_ this addresses all the cases, if one assumes:
   * Upon revoking a commitment, one sets their persistent commitment fee to the pending fee.
   * When receiving a revocation, one sets their persistent commitment fee to the pending fee.